main.tf

```diff
@@ -25,19 +25,45 @@ resource "aws_wafv2_web_acl" "main" {
       name     = lookup(rule.value, "name")
       priority = lookup(rule.value, "priority")
 
-      override_action {
-        dynamic "none" {
-          for_each = length(lookup(rule.value, "override_action", {})) == 0 || lookup(rule.value, "override_action", {}) == "none" ? [1] : []
-          content {}
+      # Action block is required for geo_match, ip_set, and ip_rate_based rules
+      dynamic "action" {
+        for_each = length(lookup(rule.value, "action", {})) == 0 ? [] : [1]
+        content {
+          dynamic "allow" {
+            for_each = length(lookup(rule.value, "action", {})) == 0 || lookup(rule.value, "action", {}) == "allow" ? [1] : []
+            content {}
+          }
+
+          dynamic "count" {
+            for_each = lookup(rule.value, "action", {}) == "count" ? [1] : []
+            content {}
+          }
+
+          dynamic "block" {
+            for_each = lookup(rule.value, "action", {}) == "block" ? [1] : []
+            content {}
+          }
         }
+      }
+
+      dynamic "override_action" {
```

Comment:
As this is changing the default behaviour and this block is fully optional, you may need to update existing examples

Reply:
Updated existing examples to make sure that they have the override_action where required.